Source/WebCore/dom/Document.cpp

```diff
@@ -2357,6 +2357,14 @@ void Document::visibilityStateChanged()
             callback();
     }
     updateServiceWorkerClientData();
+
+    // https://drafts.csswg.org/css-view-transitions/#page-visibility-change-steps
+    if (hidden()) {
+        m_inboundViewTransitionParams = nullptr;
+        if (RefPtr viewTransition = activeViewTransition())
+            viewTransition->skipViewTransition(Exception { ExceptionCode::InvalidStateError, "Old view transition aborted by document becoming hidden."_s });
+    } else
+        ASSERT(!activeViewTransition());
 }
```

Comment:
Seems like @nt1m's previous patch is more complete? in the sense that it also reacts to the event of the page being made hidden.

Reply:
Mine should be doing that too. The difference is that @nt1m is a better citizen than I am, and used a visibility change listener, rather than hardcoding it into Document.

My reasoning is that a visibility change listener fires on the active view transition at the point of the visibility change, whereas the spec requires a task to be queued and the active view transition to be retrieved when the task runs.

Unfortunately, when I implemented the queued task, it causes tests to fail, and the assert to fire.

I followed up on my existing spec issue here: https://github.com/web-platform-tests/wpt/issues/47828#issuecomment-2316520253

Assuming the queueing of the task is just invalid, then @nt1m's patch is clearly better, and we should merge it (with the addition of the level 2 specific change `m_inboundViewTransitionParams = nullptr;`)